### DISCUSSION TITLE
Strip JSON field content with onRecordsListRequest hook.

### DISCUSSION THREAD

**Discussion by JustDenP** [pocketbase/pocketbase#4603] [Q&A] [answered]
Hello, I have a posts collection, and its data is stored in a JSON field named 'content'. It's a object from prosemirror editor, so its structure is flexible inside each node, here is how it looks like:
```json
{
  "content":[
    {...},
    {...},
    {...}
  ],
  "type":"doc"
}
```

Since I cannot use an excerpt method for my field as with plain text, I want to modify each response for posts list, and for each post only include 2 nodes, with specified types and use them as an excerpt.

Finally, I figure out, how to get at least the content field itself, but every time I need to map or get nested object inside of it, I'm getting only keys like 123,412,424 instead of a real object.
Is there any way how can I get the whole parsable object? 
Thank you for any help!

Here is my code:
```javascript
onRecordsListRequest((e) => {
  // Function to strip content to the first two nodes of specified types
  function stripContent(content, allowedTypes) {
    console.log(Object.keys(content)) // Returns 123,312,432 etc
    return content.filter(item => allowedTypes.includes(item.type)).slice(0, 2);
  }

  if (e.result.totalItems > 0) {
    // Process each post
    e.result.items.forEach((post, index) => {
      // unmarshal a single json field value into the provided result
      const content = new DynamicModel({
        content: [],
        type: "",
      })
      post.unmarshalJSONField("content", content)

      if (index === 0) {
        // We have type and content here, nice!
        console.log(Object.keys(content))
      }

      // Check if the contentObject has a content property and it is an array
      if (content.content && Array.isArray(content.content)) {
        // Strip the content to the first two nodes of paragraph, bulletList, or orderedList
        post.content = stripContent(content.content, ['paragraph', 'bulletList', 'orderedList']);
      } else {
        console.error('Content array not found');
      }
    });
  }
}, "posts")
```

**Comment by ganigeorgiev** [comment, marked as answer]
The dynamic model has no information about the nested fields to create a corresponding typed struct.

For such cases you can simple use `const content = JSON.parse(post.get("content"))` and apply the filtering you need. `content` in this case will be a regular object.

_Couple additional suggestions:_
- Remove the `if (e.result.totalItems > 0) {` wrapping as this will not work when `skipTotal=1` is passed to the list request. Iterating directly `e.result.items.forEach` should be safe as `e.result.items` should be always an array even if there are no items.
- Note that the Admin UI uses the same web APIs so if you wan to apply this only for your userland requests you can add at the top of the hook something like:
   ```js
   if (e.httpContext.get("admin")) {
       return // no extra processing needed
   }
   ...
   ```

**Reply by JustDenP** [reply]
Thank you! I modified my function, the content stripped, but now I want to return the modified result, but these ones do not work.
What am I doing wrong here?

```javascript
onRecordsListRequest((e) => {
  if (e.httpContext.get("admin")) {
    return // no extra processing needed
  }

  function stripContent(content, allowedTypes) {
    return content.filter(item => allowedTypes.includes(item.type)).slice(0, 2);
  }

  e.result.items.forEach((post) => {
    const content = JSON.parse(post.get("content"))

    if (content.content && Array.isArray(content.content)) {
      post.set("content", stripContent(content.content, ['paragraph', 'bulletList', 'orderedList']))
    } else {
      console.error('Content array not found');
    }
  });

  console.log(JSON.stringify(e.result.items, null, 2)); // is fine

  // Nothing works
  // e.httpContext.json(200, (e.result))
  // return e.result;
}, "posts")
```

**Reply by ganigeorgiev** [reply]
Directly modifying the items should be enough, you don't have to write explicitly the response or return anything in addition.

If that's not what you are asking, please elaborate what do you mean by "Nothing works" - is there an error, what is the return results, etc. (check the _Admin UI > Logs_ or wrap your code with `try/catch`)

If you need further help, provide sample data so that I can try to reproduce it locally.

**Reply by JustDenP** [reply]
Sorry, it was an error inside my app after modification of the returned result.
